spec/requests/api/v1/targets_spec.rb

```diff
@@ -103,4 +103,55 @@
       end
     end
   end
+
+  describe 'display list of targets' do
+    let(:user) { create(:user) }
+    let(:headers) { user.create_new_auth_token }
+    let(:parsed_response) { JSON.parse(response.body) }
+
+    subject { get api_v1_targets_path, as: :json, headers: headers }
+
+    context 'when user has targets' do
+      let(:num_of_targets) { Faker::Number.between(from: 1, to: 10) }
+      let!(:targets) { create_list(:target, num_of_targets, user: user) }
+
+      before { subject }
+
+      it 'returns status code ok' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns all the user targets' do
+        expect(parsed_response.size).to eq(num_of_targets)
+      end
+
+      it 'returns the targets id' do
+        expect(parsed_response).to all(have_key('id'))
+      end
+
+      it 'returns the targets title' do
+        expect(parsed_response).to all(have_key('title'))
+      end
+
+      it 'returns the target longitude' do
+        expect(parsed_response).to all(have_key('longitude'))
+      end
+
+      it 'returns the target latitude' do
+        expect(parsed_response).to all(have_key('latitude'))
+      end
+
+      it 'returns the target radius' do
+        expect(parsed_response).to all(have_key('radius'))
+      end
```

Comment:
we probably should also add a case for checking that the data is correct (it returns the different targets data, using match_array). Something like:
```ruby
    it 'returns all the targets' do
      expect(parsed_response.map(&:id)).to match_array(targets.map(&:id))
    end
```
might be enough